### ISSUE TITLE
"Option not found" error classified as internal

### ISSUE THREAD

**Issue by gavincabbage** [influxdata/flux#3470]
We're seeing occasional `Option not found` errors when users update tasks and remove the `option task = {...}` block from the script. This error comes from [here](https://github.com/influxdata/flux/blob/master/ast/edit/task_editor.go#L9-L28) where it is classified as an internal error. Since this is a task syntax error, and generally caused by user error, the returned error should be classified as invalid, rather than internal. This will prevent it from being reported as a system degradation by internal monitoring tools. 

System: Cloud 2
Flux Version: `github.com/influxdata/flux v0.103.1-0.20210202003033-17007b9a6de4`

**Comment by wolffcm** [comment]
@gavincabbage I agree that this is a misclassified error, thanks for filing this.

I would think you would want to produce a more helpful error to the tasks user, so I wonder if it would be helpful for us to instead of just returning a generic `invalid` error, to return something to make it easy to robustly detect an `option not found` condition?

```Go
var OptionNotFoundError = &flux.Error{Code: codes.Invalid, Msg: "option not found"}

func GetOption(...) (ast.Expression, error) {
  ...
  return nil, OptionNotFoundError
}
```
Then in client code you could `if err == edit.OptionNotFoundError ...` to handle the error gracefully.

**Comment by gavincabbage** [comment]
Sounds great @wolffcm, I had a similar thought! We can definitely teach the task API to do that if Flux exposes a sentinel error.

**Comment by skartikey** [comment]
Fixed with PR #3539
